libs/journeys/ui/src/libs/getTextResponseLabel/getTextResponseLabel.ts

```diff
@@ -0,0 +1,26 @@
+import { TreeBlock } from '../../libs/block'
+import { BlockFields_TextResponseBlock as TextResponseBlock } from '../../libs/block/__generated__/BlockFields'
+
+function flatten(children: TreeBlock[]): TreeBlock[] {
+  return children.reduce<TreeBlock[]>(
+    (result, item) => [...result, item, ...flatten(item.children)],
+    []
+  )
+}
+
+export function getTextResponseLabel(
+  children: TreeBlock[],
+  blockId: string
+): string | null {
```

Comment:
```suggestion
export function getTextResponseLabel(
block: TreeBlock
): string | null {
```

Rather than bringing in the children array, just take in the block itself. Check that the block is a TextResponseBlock and has a non-whitespace label(as you're already doing), and return the label. no need to find descendants and matching children